libdnf5/rpm/solv/goal_private.cpp

```diff
@@ -262,61 +268,55 @@ bool GoalPrivate::limit_installonly_packages(libdnf5::solv::IdQueue & job, Id ru
             // and negative for conflicts (conflicts with another package or dependency
             // conflicts = dependencies cannot be met).
             if (libsolv_solver.get_decisionlevel(p) > 0) {
-                q.push_back(p);
+                marked_for_install_providers_map.add_unsafe(p);
             } else {
-                Solvable * solvable = spool.id2solvable(p);
-                if (!spool.is_installed(solvable)) {
-                    available_unused_providers.push_back(solvable);
-                }
+                available_unused_providers_map.add_unsafe(p);
```

Comment:
I see that the test (`spool.is_installed(solvable)`) was removed. From the description of `libsolv_solver.get_decisionlevel(p)` it looks like a redundant, but I am not sure.

Reply:
Oh, I see the test was redundant and it is only performed later and once. Good job.